packages/vite/src/executors/dev-server/dev-server.impl.ts

```diff
@@ -27,6 +27,11 @@ export default async function* viteDevServerExecutor(
     } as InlineConfig
   );
 
+  // TODO: (nicholas) clean up type
+  if (options['production']) {
```

Comment:
This is.. weird? Is this really passed? I thought we don't have a `production` option.

Reply:
yea it actually is it should be inside `options.mode` similar to what the schema suggests.

It's not critical but this is currently was is passed.

Reply:
Yeah, if you look at the logic in our other devServers.. they usually check if like.. optimization is `true`... I think this should be `options.mode === 'production`... but... that might be... also true on the `developement` mode right now :\

Reply:
`options.mode` is actually `null` I will clean up the type it's possible that it might need some normalization.